compose-app/src/main/java/com/mapbox/maps/extension/compose/MapboxMap.kt

```diff
@@ -0,0 +1,111 @@
+package com.mapbox.maps.extension.compose
+
+import android.annotation.SuppressLint
+import androidx.compose.runtime.*
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.platform.LocalInspectionMode
+import androidx.compose.ui.viewinterop.AndroidView
+import com.mapbox.maps.*
+import com.mapbox.maps.extension.compose.gestures.applySettings
+import com.mapbox.maps.extension.style.StyleContract
+import com.mapbox.maps.extension.style.style
+import com.mapbox.maps.plugin.delegates.MapPluginProviderDelegate
+import com.mapbox.maps.plugin.gestures.generated.GesturesSettings
+import com.mapbox.maps.plugin.gestures.gestures
+
+internal class MapRootNode(
+  val mapController: MapControllable,
+  val mapPluginProvider: MapPluginProviderDelegate,
+  var gesturesState: GesturesSettings,
+  var styleState: StyleContract.StyleExtension,
+  var cameraOptions: CameraOptions
+) : MapNode {
+
+  override fun onAttached() {
+    // setup map listeners
+  }
+}
+
+
+@SuppressLint("Lifecycle")
+@Composable
+public fun MapboxMap(
+  modifier: Modifier = Modifier,
+  mapInitOptions: MapInitOptions = MapInitOptions(LocalContext.current),
```

Comment:
I've got an issue when preview map in android stuido when set context in constructor

Reply:
can you clarify why we need context in preview?

Reply:
but for the preview you will draw some composable object. maybe with default params. and in that case you'll get `LocalContext`. so it might fail your preview build.

maybe we should move setter of `MapInitOptions` after preview check and pass inside that function as optional field